### PR TITLE
Community[minor]: Update VDMS vectorstore

### PR DESCRIPTION
**Pull request by cwlacewe**
**Description:** 
- This PR exposes some functions in VDMS vectorstore, updates VDMS related notebooks, updates tests, and upgrade version of VDMS (>=0.0.20)

**Issue:** N/A

**Dependencies:** 
- Update vdms>=0.0.20

### CODE REVIEW

**Comment by cwlacewe** [comment]
@baskaryan

**Comment by cwlacewe** [comment]
@ccurme or @eyurtsev could someone please review?

**Comment by cwlacewe** [comment]
@efriis @hwchase17 @vbarda

File: libs/community/tests/integration_tests/vectorstores/test_vdms.py [hunk 1]
```diff
@@ -36,19 +38,19 @@ def vdms_client() -> vdms.vdms:
 
 
 @pytest.mark.requires("vdms")
+@pytest.mark.enable_socket
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/vectorstores/test_vdms.py:41]: Why is this needed?

File: libs/community/langchain_community/vectorstores/vdms.py [hunk 2]
```diff
@@ -76,6 +76,41 @@ def _len_check_if_sized(x: Any, y: Any, x_name: str, y_name: str) -> None:
...
+                    mkey: mval
+                    for mkey, mval in ent.items()
+                    if mval not in INVALID_METADATA_VALUE
+                }
+
+                final_res.append(
+                    (Document(page_content=txt_contents, metadata=props), distance)
+                )
+    except Exception as e:
+        print(f"Error while parsing results: {e}")  # noqa: T201
```
Comment (eyurtsev) [inline libs/community/langchain_community/vectorstores/vdms.py:110]: Could you revert the print statement? If you need to log something feel free to use logger.error or raise an exception to the user

File: libs/community/tests/integration_tests/vectorstores/test_vdms.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
     import vdms
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/vectorstores/test_vdms.py:20]: FYI there's a standard test suite for read / write API for vectorstores you can use it to verify that the vectorstore is working properly 

https://github.com/langchain-ai/langchain/blob/5f8858a490347dddb680fa40f877228ef0dbd325/libs/core/tests/unit_tests/vectorstores/test_in_memory.py#L15-L15

File: libs/community/tests/integration_tests/vectorstores/test_vdms.py (hunk 1 above)
Comment (cwlacewe) [reply libs/community/tests/integration_tests/vectorstores/test_vdms.py:41]: VDMS client uses sockets so without it all test that use our client fail:
image

File: libs/community/langchain_community/vectorstores/vdms.py (hunk 2 above)
Comment (cwlacewe) [reply libs/community/langchain_community/vectorstores/vdms.py]: Reverted to logger

File: libs/community/tests/integration_tests/vectorstores/test_vdms.py (hunk 3 above)
Comment (cwlacewe) [reply libs/community/tests/integration_tests/vectorstores/test_vdms.py:20]: @eyurtsev I fixed other issues. Can we proceed without updating the entire test suite this time. Or is this required?

**Comment by cwlacewe** [comment]
Can someone please review the requested updates? @eyurtsev? @ccurme?

**Comment by cwlacewe** [comment]
@baskaryan, @efriis, @vbarda, @hwchase17,

Can someone please review this?

File: libs/community/tests/integration_tests/vectorstores/test_vdms.py (hunk 3 above)
Comment (eyurtsev) [reply libs/community/tests/integration_tests/vectorstores/test_vdms.py:20]: Not required, but it will help speed up future reviews

File: libs/community/tests/integration_tests/vectorstores/test_vdms.py (hunk 1 above)
Comment (eyurtsev) [reply libs/community/tests/integration_tests/vectorstores/test_vdms.py:41]: sockets are only blocked for unit tests not integration tests. How are you launching the test?

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
@cwlacewe sorry for the delay.

I'll merge and we've got to do a few follow up things:
1. pytest sockets enabling should be removed
2. we should remove vdms from integration_tests pyproject toml (we'll move it entirely to extended dependencies)
